### DISCUSSION TITLE
Font size does not change with --mantine-scale

### DISCUSSION THREAD

**Discussion by kfrp** [mantinedev/mantine#5723] [Feedback]
Hi, thank you for Mantine. 👏👏👏

I am sure this is not a bug, but it went against my expectations so I wanted to check in with some Mantine experts to get some clarity.

Basically I would like to get insight on what is the intended way to size a component like Menu.

I am using Mantine's Menu in multiple places but want to make it smaller in one instance - basically smaller padding and smaller font-sizes. I know that Mantine declares font-sizes as vars, eg: `--mantine-font-size-md: calc(1rem*var(--mantine-scale))` and then uses those vars in the component CSS, and uses `--mantine-scale` in padding as well.

My expectation was that setting the `--mantine-scale` variable on an element or component would affect its font-size, which it doesn't. This is just how calcs work it seems, they are calculated right where they are declared. So I cannot scale or size a component like Menu by setting `--mantine-scale` on it because all the font-size vars have already been calculated. The padding in Mantine Menu does change, because it uses calc directly instead of a padding var.

It seems that  in order to size my menu, I have the following options:
1. override the font-sizes myself on each Menu Item, 
2. use the Styles API which would make all Menus smaller
3. redeclare the font-size variables along with `--mantine-scale` at the component's root. 

Am I missing anything? Is there a clear correct approach to take? Any insight would be greatly appreciated.

**Comment by rtivital** [comment]
`theme.scale` works as intended, you can change value in the codesandbox and see how it changes – https://codesandbox.io/p/sandbox/mantine-react-template-forked-h6s8x8?file=%2Fsrc%2FApp.tsx%3A6%2C39

**Reply by kfrp** [reply]
Hi, thanks for your reply! Sorry I think codesandbox has gone private and I am not able to access your link. Do you mean that `theme.scale` works at the app root / MantineProvider level? Right, but that would change the scale of the whole app, which I do not want to do.  I looked in the docs again but I did not find any way to set theme overrides at the component level. 

My hacky but simple workaround: I added a class called `.scaled` alongside the `:root` declaration in Mantine's CSS. This covers all the scaled variables (font-size, spacing, shadow, radius etc). When I am setting `--mantine-scale: 0.9` using the `__vars` prop on my `Menu.Dropdown` component, I also add the `.scaled` class. This ensures that all the vars are recalculated with the new scale, and I am able to use the component without overriding individual styles. It will do in certain cases. 

Any advice welcome of course. Many thanks for Mantine.

**Reply by rtivital** [reply]
It is not supported to override this on component level, only on theme level for the entire application
